packages/labs/analyzer/src/test/javascript/functions_test.ts

```diff
@@ -0,0 +1,115 @@
+/**
+ * @license
+ * Copyright 2022 Google LLC
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+import {suite} from 'uvu';
+// eslint-disable-next-line import/extensions
+import * as assert from 'uvu/assert';
+import {fileURLToPath} from 'url';
+import {getSourceFilename, languages} from '../utils.js';
+
+import {
+  createPackageAnalyzer,
+  Analyzer,
+  AbsolutePath,
+  Module,
+} from '../../index.js';
+
+for (const lang of languages) {
+  const test = suite<{
+    analyzer: Analyzer;
+    packagePath: AbsolutePath;
+    module: Module;
+  }>(`Module tests (${lang})`);
+
+  test.before((ctx) => {
+    try {
+      const packagePath = fileURLToPath(
+        new URL(`../../test-files/${lang}/functions`, import.meta.url).href
+      ) as AbsolutePath;
+      const analyzer = createPackageAnalyzer(packagePath);
+
+      const result = analyzer.getPackage();
+      const file = getSourceFilename('functions', lang);
+      const module = result.modules.find((m) => m.sourcePath === file);
+      if (module === undefined) {
+        throw new Error(`Analyzer did not analyze file '${file}'`);
+      }
+
+      ctx.packagePath = packagePath;
+      ctx.analyzer = analyzer;
+      ctx.module = module;
+    } catch (error) {
+      // Uvu has a bug where it silently ignores failures in before and after,
+      // see https://github.com/lukeed/uvu/issues/191.
+      console.error('uvu before error', error);
+      process.exit(1);
+    }
+  });
+
+  test('Function 1', ({module}) => {
+    const exportedFn = module.getResolvedExport('function1');
+    const fn = module.getDeclaration('function1');
+    assert.equal(fn, exportedFn);
+    assert.ok(fn?.isFunctionDeclaration());
+    assert.equal(fn.name, `function1`);
+    assert.equal(fn.description, `Function 1 description`);
+    assert.equal(fn.summary, undefined);
+    assert.equal(fn.parameters?.length, 0);
+    assert.equal(fn.deprecated, undefined);
+  });
+
+  test('Function 2', ({module}) => {
+    const exportedFn = module.getResolvedExport('function2');
+    const fn = module.getDeclaration('function2');
+    assert.equal(fn, exportedFn);
+    assert.ok(fn?.isFunctionDeclaration());
+    assert.equal(fn.name, `function2`);
+    assert.equal(fn.summary, `Function 2 summary\nwith wraparound`);
+    assert.equal(fn.description, `Function 2 description\nwith wraparound`);
+    assert.equal(fn.parameters?.length, 3);
+    assert.equal(fn.parameters?.[0].name, 'a');
+    assert.equal(fn.parameters?.[0].description, 'Param a description');
+    assert.equal(fn.parameters?.[0].summary, undefined);
+    assert.equal(fn.parameters?.[0].type?.text, 'string');
+    assert.equal(fn.parameters?.[0].default, undefined);
+    assert.equal(fn.parameters?.[0].rest, false);
```

Comment:
Uvu's `equal()` is a deep equal... can you use that for better readability?

Reply:
It seems to do pretty bad things with classes, so gonna avoid for now.